core/src/test/java/org/apache/iceberg/TestRowDelta.java

```diff
@@ -1134,4 +1135,121 @@ public void testConcurrentNonConflictingRowDelta() {
 
     validateTableDeleteFiles(table, deleteFile1, deleteFile2);
   }
+
+  @Test
+  public void testConcurrentNonConflictingRowDeltaAndRewriteFilesWithSequenceNumber() {
+    // change the spec to be partitioned by data
+    table.updateSpec()
+        .removeField(Expressions.bucket("data", 16))
+        .addField(Expressions.ref("data"))
+        .commit();
+
+    // add a data file to partition A
+    DataFile dataFile1 = DataFiles.builder(table.spec())
+        .withPath("/path/to/data-a.parquet")
+        .withFileSizeInBytes(10)
+        .withPartitionPath("data=a")
+        .withRecordCount(1)
+        .build();
+
+    table.newAppend()
+        .appendFile(dataFile1)
+        .commit();
+
+    Snapshot baseSnapshot = table.currentSnapshot();
+
+    // add an equality delete file
+    DeleteFile deleteFile1 = FileMetadata.deleteFileBuilder(table.spec())
```

Comment:
We have some helpers for this (at least with data files)

```newDeleteFile and newDataFile```

In testTableBase